src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -936,6 +949,9 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
         snapshotHeader.stateRoot.toBuffer()
       );
       blocks.latest = snapshotBlock;
+
+      // notify subscribers of reorg
+      await this.emit("block", snapshotBlock);
```

Comment:
Maybe I'm misunderstanding the concern, but I think the order would still need to be `blockLogs` and then `block`, for the same reason that you described? If the order were reversed, the application code might see a rollback on `block` first, but the `blockLogs` with the `removed` flag set might not be present in their filter.

I wonder how often this occurs in practice, however. I'd expect if someone is using pub/sub they'd use a logs subscription rather than `eth_getFilterLogs`, but I'm sure there's some code out there that does this, as devs generally aren't super disciplined about swapping things over all at once when doing things like moving from polling to pub/sub.